asmcomp/x86_proc.ml

```diff
@@ -83,9 +83,12 @@ let string_of_symbol prefix s =
     Buffer.add_string b prefix;
     String.iter
       (function
-        | ('A'..'Z' | 'a'..'z' | '0'..'9' | '_' | '.') as c ->
+        | ('A'..'Z' | 'a'..'z' | '0'..'9' | '_') as c ->
```

Comment:
Don't you need to update the logic in the for-loop above responsible for setting the `spec` ref.

Reply:
Oops, indeed. Good catch!